src/validate.ts

```diff
@@ -0,0 +1,110 @@
+'use-strict';
+
+/**
+ * Generates random request ID 
+ * @param min Minimum number
+ * @param max Maximum number 
+ * @returns Random number between the range of the min to max number specified
+ */
+function generateRequestID(min, max){
+     return Math.floor(Math.random() * (max-min) + min).toString()
+}
+    
+/**
+ * Generates SYNC request 
+ * @returns Specified format for SYNC intent request. 
+ */
+export function generateSyncRequest(){
+    const requestId = generateRequestID(100,999);
+    return {
+        requestId,
+        inputs: [{
+          intent: "action.devices.SYNC"
+        }]
+    }
+}
+
+type CustomData = {[key: string]: any}
```

Comment:
I would recommend you bring this together into a single `Device` interface, e.g.

```typescript
interface Device {
  id: string,
  customData?: {[key: string]: any},
}
```

Reply:
Revised!